### ISSUE TITLE
Keep running into authentication errors

### ISSUE THREAD

**Issue by springcoil** [cbetz/flutter-vision#2]
Hi. 

I keep running into errors like this. Do you have any insight?

```   ] E/StorageException(15783): 	at com.google.firebase.storage.zzh.run(Unknown Source:2)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
[        ] E/StorageException(15783): 	at java.lang.Thread.run(Thread.java:764)
[        ] E/StorageException(15783): Caused by: java.io.IOException: {  "error": {    "code": 403,    "message": "Permission denied. Could not perform this operation"  }}
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:3435)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:65)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:57)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzb(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzc(com.google.firebase:firebase-storage@@16.0.2:4255)
[        ] E/StorageException(15783): 	... 5 more
[        ] E/StorageException(15783): StorageException has occurred.
[        ] E/StorageException(15783): User does not have permission to access this object.
[        ] E/StorageException(15783):  Code: -13021 HttpResult: 403
[  +19 ms] E/StorageException(15783): The server has terminated the upload session
[        ] E/StorageException(15783): java.io.IOException: The server has terminated the upload session
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzj(com.google.firebase:firebase-storage@@16.0.2:318)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzi(com.google.firebase:firebase-storage@@16.0.2:287)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzc(com.google.firebase:firebase-storage@@16.0.2:201)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.StorageTask.zza(com.google.firebase:firebase-storage@@16.0.2:1056)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.zzh.run(Unknown Source:2)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
[        ] E/StorageException(15783): 	at java.lang.Thread.run(Thread.java:764)
[        ] E/StorageException(15783): Caused by: java.io.IOException: {  "error": {    "code": 403,    "message": "Permission denied. Could not perform this operation"  }}
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:3435)
[   +7 ms] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:65)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:57)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzb(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzc(com.google.firebase:firebase-storage@@16.0.2:4255)
[        ] E/StorageException(15783): 	... 5 more
[        ] E/StorageException(15783): StorageException has occurred.
[        ] E/StorageException(15783): User does not have permission to access this object.
[        ] E/StorageException(15783):  Code: -13021 HttpResult: 403
[        ] E/StorageException(15783): The server has terminated the upload session
[        ] E/StorageException(15783): java.io.IOException: The server has terminated the upload session
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzj(com.google.firebase:firebase-storage@@16.0.2:318)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzi(com.google.firebase:firebase-storage@@16.0.2:287)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzc(com.google.firebase:firebase-storage@@16.0.2:201)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.StorageTask.zza(com.google.firebase:firebase-storage@@16.0.2:1056)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.zzh.run(Unknown Source:2)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
[        ] E/StorageException(15783): 	at java.lang.Thread.run(Thread.java:764)
[        ] E/StorageException(15783): Caused by: java.io.IOException: {  "error": {    "code": 403,    "message": "Permission denied. Could not perform this operation"  }}
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:3435)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:65)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:57)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzb(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzc(com.google.firebase:firebase-storage@@16.0.2:4255)
[        ] E/StorageException(15783): 	... 5 more
[        ] E/StorageException(15783): StorageException has occurred.
[        ] E/StorageException(15783): User does not have permission to access this object.
[        ] E/StorageException(15783):  Code: -13021 HttpResult: 403
[        ] E/StorageException(15783): The server has terminated the upload session
[        ] E/StorageException(15783): java.io.IOException: The server has terminated the upload session
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzj(com.google.firebase:firebase-storage@@16.0.2:318)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzi(com.google.firebase:firebase-storage@@16.0.2:287)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzc(com.google.firebase:firebase-storage@@16.0.2:201)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.StorageTask.zza(com.google.firebase:firebase-storage@@16.0.2:1056)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.zzh.run(Unknown Source:2)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
[        ] E/StorageException(15783): 	at java.lang.Thread.run(Thread.java:764)
[        ] E/StorageException(15783): Caused by: java.io.IOException: {  "error": {    "code": 403,    "message": "Permission denied. Could not perform this operation"  }}
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:3435)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:65)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:57)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzb(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.UploadTask.zzc(com.google.firebase:firebase-storage@@16.0.2:4255)
[        ] E/StorageException(15783): 	... 5 more
[        ] I/flutter (15783): EVENT StorageTaskEventType.failure
[   +1 ms] E/StorageUtil(15783): error getting token java.util.concurrent.ExecutionException: com.google.firebase.FirebaseApiNotAvailableException: firebase-auth is not linked,
please fall back to unauthenticated mode.
[        ] W/NetworkRequest(15783): no auth token for request
[ +349 ms] E/StorageException(15783): StorageException has occurred.
[        ] E/StorageException(15783): User does not have permission to access this object.
[        ] E/StorageException(15783):  Code: -13021 HttpResult: 403
[        ] E/StorageException(15783): {  "error": {    "code": 403,    "message": "Permission denied. Could not perform this operation"  }}
[        ] E/StorageException(15783): java.io.IOException: {  "error": {    "code": 403,    "message": "Permission denied. Could not perform this operation"  }}
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:455)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzj.zza(com.google.firebase:firebase-storage@@16.0.2:3435)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:65)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.obfuscated.zzc.zza(com.google.firebase:firebase-storage@@16.0.2:57)
[        ] E/StorageException(15783): 	at com.google.firebase.storage.zzc.run(com.google.firebase:firebase-storage@@16.0.2:68)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
[        ] E/StorageException(15783): 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
[        ] E/StorageException(15783): 	at java.lang.Thread.run(Thread.java:764)
[   +9 ms] E/flutter (15783): [ERROR:flutter/shell/common/shell.cc(184)] Dart Error: Unhandled exception:
[        ] E/flutter (15783): PlatformException(download_error, User does not have permission to access this object., null)
[        ] E/flutter (15783): #0      StandardMethodCodec.decodeEnvelope (package:flutter/src/services/message_codecs.dart:551:7)
[        ] E/flutter (15783): #1      MethodChannel.invokeMethod (package:flutter/src/services/platform_channel.dart:292:18)
[        ] E/flutter (15783): <asynchronous suspension>
[        ] E/flutter (15783): #2      StorageReference.getDownloadURL
(file:///Users/peadarcoyle/development/flutter/.pub-cache/hosted/pub.dartlang.org/firebase_storage-1.0.4/lib/src/storage_reference.dart:136:10)
[        ] E/flutter (15783): <asynchronous suspension>
[        ] E/flutter (15783): #3      _CameraScreenState._uploadFile (package:flutter_vision/main.dart:166:65)
[        ] E/flutter (15783): <asynchronous suspension>
[        ] E/flutter (15783): #4      _CameraScreenState.detectLabels (package:flutter_vision/main.dart:139:38)
[        ] E/flutter (15783): <asynchronous suspension>
[        ] E/flutter (15783): #5      _CameraScreenState.onTakePictureButtonPressed.<anonymous closure> (package:flutter_vision/main.dart:118:11)
[        ] E/flutter (15783): #6      _RootZone.runUnary (dart:async/zone.dart:1379:54)
[        ] E/flutter (15783): #7      _FutureListener.handleValue (dart:async/future_impl.dart:129:18)
[        ] E/flutter (15783): #8      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:642:45)
[        ] E/flutter (15783): #9      Future._propagateToListeners (dart:async/future_impl.dart:671:32)
[        ] E/flutter (15783): #10     Future._complete (dart:async/future_impl.dart:476:7)
[        ] E/flutter (15783): #11     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
[        ] E/flutter (15783): #12     _AsyncAwaitCompleter.complete (dart:async/runtime/libasync_patch.dart:28:18)
[        ] E/flutter (15783): #13     _completeOnAsyncReturn (dart:async/runtime/libasync_patch.dart:295:13)
[        ] E/flutter (15783): #14     _CameraScreenState.takePicture (package:flutter_vision/main.dart)
[        ] E/flutter (15783): <asynchronous suspension>
[        ] E/flutter (15783): #15     _CameraScreenState.onTakePictureButtonPressed (package:flutter_vision/main.dart:112:5)
[        ] E/flutter (15783): #16     _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:507:14)
[        ] E/flutter (15783): #17     _InkResponseState.build.<anonymous closure> (package:flutter/src/material/ink_well.dart:562:30)
[        ] E/flutter (15783): #18     GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:102:24)
[        ] E/flutter (15783): #19     TapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:242:9)
[        ] E/flutter (15783): #20     TapGestureRecognizer.handlePrimaryPointer (package:flutter/src/gestures/tap.dart:175:7)
[        ] E/flutter (15783): #21     PrimaryPointerGestureRecognizer.handleEvent (package:flutter/src/gestures/recognizer.dart:315:9)
[        ] E/flutter (15783): #22     PointerRouter._dispatch (package:flutter/src/gestures/pointer_router.dart:73:12)
[        ] E/flutter (15783): #23     PointerRouter.route (package:flutter/src/gestures/pointer_router.dart:101:11)
[        ] E/flutter (15783): #24     _WidgetsFlutterBinding&BindingBase&GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:180:19)
[        ] E/flutter (15783): #25     _WidgetsFlutterBinding&BindingBase&GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:158:22)
[        ] E/flutter (15783): #26     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerEvent (package:flutter/src/gestures/binding.dart:138:7)
[        ] E/flutter (15783): #27     _WidgetsFlutterBinding&BindingBase&GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:101:7)
[        ] E/flutter (15783): #28     _WidgetsFlutterBinding&BindingBase&GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:85:7)
[        ] E/flutter (15783): #29     _invoke1 (dart:ui/hooks.dart:168:13)
[        ] E/flutter (15783): #30     _dispatchPointerDataPacket (dart:ui/hooks.dart:122:5)
[+13487 ms] I/zygote64(15783): Do full code cache collection, code=123KB, data=88KB
[        ] I/zygote64(15783): After code cache collection, code=112KB, data=63KB```

**Comment by springcoil** [comment]
It looks to me like an authentication error, I've tried a few times. I notice for example nothing is created in my database

**Comment by springcoil** [comment]
Found solution here https://github.com/distriqt/ANE-Firebase/issues/160
TLDR; is ```in the firebase storage rules i had to take out if request.auth != null;
allow read, write;```
